sonic_y_cable/y_cable.py

```diff
@@ -10,8 +10,8 @@
     import struct
     from ctypes import c_int8
 
-    import sonic_platform.platform
     from sonic_py_common import logger
+    import sonic_platform.platform
```

Comment:
I don't understand why this order makes a difference. You state in the PR description,

>Import of sonic_platform before logger when platform API is not available leads to non import of logger resulting in xcvrd crash

Where does the "non import of logger" occur?

Reply:
In case platform API is not available, it results in the import of logger in the subsequent line not being performed and thus accessing it later as below raises an Exception:   
https://github.com/Azure/sonic-platform-common/blob/10dc16f40baa0867da21fb31c089b73d33cc4107/sonic_y_cable/y_cable.py#L110

Reply:
Under what circumstances is the platform API not available? I think maybe we need to rethink/refactor something here, rather than simply change the order of imports.

Reply:
This issue is seen in platforms, in which the platform API is currently not available/implemented.

Reply:
I see. I was hoping this was the answer. As of now, we expect all platforms to have sonic-platform package support, rather than the old plugins. So I'm OK with taking this change for now to give vendors a bit more time to move over, but this is not the standard import ordering, so I intend to change it back in the future.